assets/app/views/builds.html

```diff
@@ -19,10 +19,23 @@
           </div>
           <div ng-switch="buildConfig.parameters.strategy.type">
             <div ng-switch-when="STI">
-              <dl class="dl-horizontal">
-                <dt>Builder image:</dt>
-                <dd>{{buildConfig.parameters.strategy.stiStrategy.image}}</dd>
-              </dl>
+              <div ng-if="buildConfig.parameters.strategy.stiStrategy.image">
+                <dl class="dl-horizontal">
+                  <dt>Builder image:</dt>
+                  <dd>{{buildConfig.parameters.strategy.stiStrategy.image}}</dd>
+                </dl>
+              </div>
+              <div ng-if="buildConfig.parameters.strategy.stiStrategy.from">
+                <dl class="dl-horizontal">
+                  <dt>Builder image repo:</dt>
+                  <div ng-if="buildConfig.parameters.strategy.stiStrategy.from.namespace">
+                    <dd>{{buildConfig.parameters.strategy.stiStrategy.from.namespace}}/{{buildConfig.parameters.strategy.stiStrategy.from.name}}</dd>
```

Comment:
You can put the ng-if condition in the &lt;dd&gt; tag itself and remove the extra div
If there's a remote chance that we will display object references like this elsewhere in the UI, I'd say that the logic should go in a filter. {{ buildConfig.parameters.strategy.stiStrategy.from | objectReference }}